pkg/reconciler/instances/istio/README.md

```diff
@@ -30,3 +30,33 @@ Follow these steps to run Istio Reconciler on your local machine:
    ```bash
     ./bin/mothership-darwin local --value global.ingress.domainName=example.com,global.domainName=example.com --components cluster-essentials,istio
    ```
+
+## Details
+
+Reconciliation in Kyma is handled by Reconciler. The Mothership Reconciler knows the reconciliation status of every managed Kyma cluster and initiates reconciliation of all Kyma components.
+
+The reconciliation of Istio is split into two different Kyma components: Istio and Istio Resources. The separation enables having the pure installation process of Istio under control via Istio Reconciler implementation and the Reconciler framework. The Reconciler framework functionality simply applies charts to the cluster for Istio Resources.
+
+The Istio component installs Istio on a Kyma cluster. For installation purposes, it requires Reconciler that uses `istioctl` and a rendered `istio-operator.yaml` file with Kyma-specific configurations. After proper installation/upgrade of the Istio service mesh within the cluster, the Istio proxies are restarted if needed. Within the whole Kyma reconciliation process, the installation of Istio is a prerequisite as other components rely on the proper service mesh setup and on the installed CRDs.
+
+The Istio Resources component provides charts for additional resources that are related to Istio itself but are not related to the installation process. The resources are as follows:
+
+- Istio monitoring configuration details that provide Grafana dashboards specification
+- Istio Ingress Gateway that configures incoming traffic to Kyma
+- Mutual TLS (mTLS) configuration that enables mTLS cluster-wide in the STRICT mode
+- Service Monitor that configures monitoring for the Istio component
+- Istio [VirtualService](https://istio.io/docs/reference/config/networking/virtual-service/) that informs whether Istio is up and running
+
+See the diagram for details:
+
+![Istio Reconciler](./assets/istio-reconciler.svg)
+
+This activity diagram presents detailed steps of the Istio installation/update process:
+
+![Istio Reconciliation](./assets/istio-reconciliation-action.svg)
+
+The reconciliation is executed by the Istio Reconciler. First, it verifies, using the rules explained in the diagram, if the Istio version found on the cluster and the Client version (istioctl) match. If the versions are compatible, either an installation or update process is triggered. Before the update, the version from the Istio [`values.yaml`](https://github.com/kyma-project/kyma/blob/main/resources/istio-configuration/values.yaml) is compared with the cluster version.
```

Comment:
```suggestion
The reconciliation is executed by the Istio Reconciler. First, it verifies, using the rules explained in the diagram, whether the Istio version found on the cluster and the Client version (istioctl) match. If the versions are compatible, either an installation or update process is triggered. Before the update, the version from the Istio [`values.yaml`](https://github.com/kyma-project/kyma/blob/main/resources/istio-configuration/values.yaml) is compared with the cluster version.
```
I'd put the "Before the update..." sentence before the "...update" sentence, as that step happens before?

Maybe a numbered list would clarify the steps, let's see

```suggestion
The reconciliation is executed by the Istio Reconciler. 

1. Istio reconciler verifies whether the Istio version found on the cluster and the Client version (istioctl) match. See the diagram for details on the criteria.
   If the versions are compatible, either an installation or update process is possible.
2. Before the update, the version from the Istio [`values.yaml`](https://github.com/kyma-project/kyma/blob/main/resources/istio-configuration/values.yaml) is compared with the cluster version.
3. The installation or update is triggered.
```